manim/__main__.py

```diff
@@ -1,14 +1,178 @@
+    try:
+        user_input = input(constants.CHOOSE_NUMBER_MESSAGE)
+        return [
+            num_to_class[int(num_str)]
+            for num_str in user_input.split(",")
+        ]
+    except KeyError:
+        logger.error(constants.INVALID_NUMBER_MESSAGE)
+        sys.exit(2)
+        user_input = input(constants.CHOOSE_NUMBER_MESSAGE)
+        return [
+            num_to_class[int(num_str)]
+            for num_str in user_input.split(",")
+        ]
+    except EOFError:
+        sys.exit(1)
+
+
+def get_scenes_to_render(scene_classes):
+    if not scene_classes:
+        logger.error(constants.NO_SCENE_MESSAGE)
+        return []
+    if file_writer_config["write_all"]:
+        return scene_classes
+    result = []
+    for scene_name in file_writer_config["scene_names"]:
+        found = False
+        for scene_class in scene_classes:
+            if scene_class.__name__ == scene_name:
+                result.append(scene_class)
+                found = True
+                break
+        if not found and (scene_name != ""):
+            logger.error(
+                constants.SCENE_NOT_FOUND_MESSAGE.format(
+                    scene_name
+                )
+            )
+    if result:
+        return result
+    return [scene_classes[0]] if len(scene_classes) == 1 else prompt_user_for_choice(scene_classes)
+
+
+def get_scene_classes_from_module(module):
+    if hasattr(module, "SCENES_IN_ORDER"):
+        return module.SCENES_IN_ORDER
+    else:
+        return [
+            member[1]
+            for member in inspect.getmembers(
+                module,
+                lambda x: is_child_scene(x, module)
+            )
+        ]
+
+
+def get_module(file_name):
+    if file_name == "-":
+        module = types.ModuleType("input_scenes")
+        code = sys.stdin.read()
+        try:
+            exec(code, module.__dict__)
+            return module
+        except Exception as e:
+            logger.error(f"Failed to render scene: {str(e)}")
+            sys.exit(2)
+    else:
+        if os.path.exists(file_name):
+            module_name = file_name.replace(os.sep, ".").replace(".py", "")
+            spec = importlib.util.spec_from_file_location(module_name, file_name)
+            module = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(module)
+            return module
+        else:
+            raise FileNotFoundError(f'{file_name} not found')
 
 
 def main():
-    args = config.parse_cli()
-    cfg = config.get_configuration(args)
-    config.initialize_directories(cfg)
-    config.initialize_tex(cfg)
-    extract_scene.main(cfg)
+    module = get_module(file_writer_config["input_file"])
+    all_scene_classes = get_scene_classes_from_module(module)
+    scene_classes_to_render = get_scenes_to_render(all_scene_classes)
+
+    for SceneClass in scene_classes_to_render:
+        try:
+            # By invoking, this renders the full scene
+            scene = SceneClass(**{})
```

Comment:
Is there any reason this can't just be `SceneClass()`?

Reply:
Brainfart is the reason